config/base/src/lib.rs

```diff
@@ -300,54 +299,82 @@ pub mod derive {
-        #[error("Failed to (de)serialize the field: {}", .0.field)]
+        #[error("Failed to (de)serialize the field: {}", .field)]
         #[serde(skip)]
-        FieldError(#[from] FieldError),
-        /// Used in [`Builder`] trait for build errors
-        #[error("Proxy failed at build stage due to: {0}")]
-        ProxyBuildError(String),
-        /// Used in the [`LoadFromDisk`](`crate::proxy::LoadFromDisk`) trait for file read errors
-        #[error("Reading file from disk failed: {0}")]
+        FieldDeserialization {
+            /// Field name (known at compile time)
+            field: &'static str,
+            /// Serde json error
+            #[source]
+            error: serde_json::Error,
+        },
+
+        /// When a field is missing.
+        #[error("Please add `{}` to the configuration.", .field)]
         #[serde(skip)]
-        DiskError(#[from] std::io::Error),
-        /// Used in [`LoadFromDisk`](`crate::proxy::LoadFromDisk`) trait for deserialization errors
-        #[error("Deserializing JSON failed: {0}")]
+        MissingField {
+            /// Field name
+            field: &'static str,
+            /// Additional message to be added as `color_eyre::suggestion`
+            message: &'static str,
+        },
+
+        /// Key pair creation failed, most likely because the keys don't form a pair
+        #[error("Key pair creation failed")]
+        Crypto(#[from] iroha_crypto::Error),
+
+        // IMO this variant should not exist. If the value is inferred, we should only warn people if the inferred value is different from the provided one.
+        /// Inferred field was provided by accident and we don't want it to be provided, because the value is inferred from other fields
+        #[error("You should remove the field `{}` as its value is determined by other configuration parameters.", .field)]
         #[serde(skip)]
-        SerdeError(#[from] json5::Error),
-    }
+        ProvidedInferredField {
+            /// Field name
+            field: &'static str,
+            /// Additional message to be added as `color_eyre::suggestion`
+            message: &'static str,
+        },
 
-    impl Error {
-        /// Construct a field error
-        pub const fn field_error(field: &'static str, error: serde_json::Error) -> Self {
-            Self::FieldError(FieldError { field, error })
-        }
+        /// Value that is unacceptable to Iroha was encountered when deserializing the config
+        #[error("The value {} of {} is wrong. \nPlease change the value.", .value, .field)]
+        #[serde(skip)]
+        InsaneValue {
+            /// The value of the field that's incorrect
+            value: String,
+            /// Field name that contains invalid value
+            field: &'static str,
+            /// Additional message to be added as `color_eyre::suggestion`
+            message: String,
+            // docstring: &'static str,  // TODO: Inline the docstring for easy access
```

Comment:
Or maybe generating a link such as `iroha/docs/source/references/config.md#queuetransaction_time_to_live_ms` would be easier?

Reply:
Created #3326